R/convertToPhyloseq.R

```diff
@@ -0,0 +1,264 @@
+#'   included in the phyloseq object that is created. 
+#'   (By default: \code{assay.type = "counts"})
+#'   
+#' @param assay_name a single \code{character} value for specifying which
+#'   assay to use for calculation.
+#'   (Please use \code{assay.type} instead. At some point \code{assay_name}
+#'   will be disabled.)
+#'   
+#' @param tree.name a single \code{character} value for specifying which
+#'   tree will be included in the phyloseq object that is created, 
+#'   (By default: \code{tree.name = "phylo"})
+#'   
+#' @param tree_name a single \code{character} value for specifying which
+#'   tree will be included in the phyloseq object that is created, 
+#'   (By default: \code{tree_name = "phylo"})
+#'   \code{tree_name} will be deprecated. Use \code{tree.name} instead.
+#' 
+#' @details 
+#' \code{convertToPhyloseq} creates a phyloseq object from a 
+#' \code{\link[TreeSummarizedExperiment:TreeSummarizedExperiment-class]{TreeSummarizedExperiment}}
+#' object. By using \code{assay.type}, it is possible to specify which table
+#' from \code{assay} is added to the phyloseq object.
+#'
+#' @return
+#' \code{convertToPhyloseq} returns an object of class 
+#' \code{\link[phyloseq:phyloseq-class]{phyloseq}}
+#'
+#' @author Leo Lahti and Tuomas Borman. Contact: \url{microbiome.github.io}
+#'
+#' @examples
+#' 
+#' ### Coerce a TreeSE object to a phyloseq object
+#' # Get tse object
+#' data(GlobalPatterns)
+#' tse <- GlobalPatterns
+#'
+#' # Create a phyloseq object from it
+#' phy <- convertToPhyloseq(tse)
+#' phy
+#'
+#' # By default the chosen table is counts, but if there are other tables,
+#' # they can be chosen with assay.type.
+#'
+#' # Counts relative abundances table
+#' tse <- transformAssay(tse, method = "relabundance")
+#' phy2 <- convertToPhyloseq(tse, assay.type = "relabundance")
+#' phy2
+#' @rdname convert
+#' @export
+setGeneric("convertToPhyloseq", signature = c("x"),
+           function(x, ...)
+             standardGeneric("convertToPhyloseq"))
+
+#' @rdname convert
+#' @export
+setMethod("convertToPhyloseq",
+          signature = c(x = "SummarizedExperiment"),
+    function(x, assay.type = "counts", assay_name = NULL, ...){
+      # Input check
+      .require_package("phyloseq")
+      # Check that tse do not have zero rows
+      if(!all(dim(x) > 0)){
+        stop("'x' contains zero rows. 'x' can not be converted
+             to a phyloseq object.",
+             call. = FALSE)
+      }
+      
+      if (!is.null(assay_name)) {
+        .Deprecated(old="assay_name", new="assay.type", "Now assay_name is deprecated. Use assay.type instead.")
+      }
+      
+      # Check assay.type
+      .check_assay_present(assay.type, x)
+      
+      # phyloseq object requires nonduplicated rownames. If there are 
+      # duplicated rownames, they are converted so that they are unique
+      if( any(duplicated(rownames(x))) ){
+        rownames(x) <- getTaxonomyLabels(x)
+      }
+      # List of arguments
```

Comment:
These look like they are new functions even though they are not. The name has only changed. This cannot be merged since this breaks the history of the functions. One option is to create new branch and continue from there

Reply:
OK I think I am going to create a new branch then, it will be easier